File: Sorting/BubbleSort.java
import java.util.*;
// worst case O(n) ^ 2
// best case O(n)
public class BubbleSort extends SortParent {
    public static void main(String[] args) {
        BubbleSort bs = new BubbleSort();
        bs.sort();
    }

    /**
     * buble sort is n^2 at worst case
     * then it's n in it's best case
     */
    public int[] sort() {
        int[] answers = numbers;
        int n = numbers.length;
        for (int i = 0; i <  n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (numbers[j] > numbers[j + 1]) {
                    int temp = numbers[j];
                    numbers[j] = numbers[j + 1];
                    numbers[j + 1] = temp;
                    printArray();
                }
            }
        }
        return answers;
    }
}

File: Sorting/bubble-sort.js
const bubbleSort = arr => {
    var n = arr.length;
    for (let i = 0; i < n; i++) {
        for (let j = 0;j < n -i -1; j++) {
            if (arr[j] > arr[j + 1]) {
                var temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
                console.log(arr);
            }
        }
    }
    return arr;
}


const arr = [10, 9, 8, 7, 6, 5, 4, 3, 2, 1];
console.log(bubbleSort(arr));
